domain/ide/hover/service.go

```diff
@@ -23,27 +23,30 @@ import (
 
 	"github.com/snyk/snyk-ls/application/config"
 	"github.com/snyk/snyk-ls/domain/snyk"
+	"github.com/snyk/snyk-ls/internal/product"
 )
 
 type Service interface {
-	DeleteHover(path string)
+	DeleteHover(p product.Product, path string)
 	Channel() chan DocumentHovers
 	ClearAllHovers()
 	GetHover(path string, pos snyk.Position) Result
 }
 
+type hoversByProduct map[product.Product][]Hover[Context]
+
 type DefaultHoverService struct {
-	hovers       map[string][]Hover[Context]
-	hoverIndexes map[string]bool
-	hoverChan    chan DocumentHovers
-	mutex        *sync.RWMutex
-	c            *config.Config
+	hoversByFilePath map[string]hoversByProduct
```

Comment:
note: we should introduce at a later time a type alias for string that is used for paths (path/fileSystemPath)

Reply:
alternative: golang URIs